benchmarks/util.hpp

```diff
@@ -0,0 +1,20 @@
+#include <random>
+#include <vector>
+#include <algorithm>
+
+template <typename T>
+std::vector<T> make_vector(size_t size)
+{
+    std::random_device rd;  
+    std::mt19937 mersenne(rd());
+    std::uniform_real_distribution<T> dist(0.0, 1.0);
```

Comment:
Can we keep the distribution / generator in `double`, and only assign to `T` in the vector in the end? There's no need to have the random generator itself work with active types.

Also this function should better be declared `inline`, as it is in a header.